### PR TITLE
Various fixes to varLib.interpolatable

### PR DESCRIPTION
**Pull request by simoncozens**
* Allow UFO files to be checked
* Report open paths (instead of raising an exception)
* Report non-existant glyphs (instead of raising an exception)
* Reformat code with black
* Tidy up output and add an option (--json) to produce a machine-readable list of incompatibilities

Example text output:

```
Glyph D was not compatible:
    Node 1 incompatible in path 0: curveTo in Coolangatta-Regular, lineTo in Coolangatta-Bold
    Node count differs in path 1: 7 in Coolangatta-Regular, 6 in Coolangatta-Bold
Glyph N was not compatible:
    Node 11 incompatible in path 0: curveTo in Coolangatta-Regular, lineTo in Coolangatta-Bold
Glyph v was not compatible:
    Node 7 incompatible in path 0: lineTo in Coolangatta-Regular, curveTo in Coolangatta-Bold
```

Example JSON output:

```JSON
{
  "D": [
    {
      "type": "node_incompatibility",
      "path": 0,
      "node": 1,
      "master_1": "Coolangatta-Regular",
      "master_2": "Coolangatta-Bold",
      "value_1": "curveTo",
      "value_2": "lineTo"
    },
    {
      "type": "node_count",
      "path": 1,
      "master_1": "Coolangatta-Regular",
      "master_2": "Coolangatta-Bold",
      "value_1": 7,
      "value_2": 6
    }
  ],
  "N": [
    {
      "type": "node_incompatibility",
      "path": 0,
      "node": 9,
      "master_1": "Coolangatta-Regular",
      "master_2": "Coolangatta-Bold",
      "value_1": "curveTo",
      "value_2": "lineTo"
    }
  ],
  "v": [
    {
      "type": "node_incompatibility",
      "path": 0,
      "node": 7,
      "master_1": "Coolangatta-Regular",
      "master_2": "Coolangatta-Bold",
      "value_1": "lineTo",
      "value_2": "curveTo"
    }
  ]
}
```

### CODE REVIEW

**Comment by anthrotype** [comment]
can you please try to rebase off master? I just merged GH Actions PR

**Review by anthrotype — APPROVED** [review]
This is very useful, thanks Simon!

File: Lib/fontTools/varLib/interpolatable.py [hunk 1]
```diff
@@ -121,7 +121,11 @@ def test(glyphsets, glyphs=None, names=None):
 				for contour in contourPens:
 					nodeTypes.append(tuple(instruction[0] for instruction in contour.value))
 					stats = StatisticsPen(glyphset=glyphset)
-					contour.replay(stats)
+					try:
+						contour.replay(stats)
+					except NotImplementedError as e:
```
[the remark this answers is not in the document]
Comment (simoncozens) [reply Lib/fontTools/varLib/interpolatable.py]: It's in MomentsPen.

[the remark this answers is not in the document]
Comment (anthrotype) [reply Lib/fontTools/varLib/interpolatable.py]: I suggest you raise a newly defined `OpenContourError` from `MomentsPen` and catch that from `interpolatable` module, e.g.:

```diff
diff --git a/Lib/fontTools/pens/momentsPen.py b/Lib/fontTools/pens/momentsPen.py
index bc1df94c..9ed08e85 100644
--- a/Lib/fontTools/pens/momentsPen.py
+++ b/Lib/fontTools/pens/momentsPen.py
@@ -7,6 +7,11 @@ from fontTools.pens.basePen import BasePen
 __all__ = ["MomentsPen"]


+class OpenContourError(NotImplementedError):
+       pass
+
+
+
 class MomentsPen(BasePen):

        def __init__(self, glyphset=None):
@@ -30,8 +35,9 @@ class MomentsPen(BasePen):
        def _endPath(self):
                p0 = self._getCurrentPoint()
                if p0 != self.__startPoint:
-                       # Green theorem is not defined on open contours.
-                       raise NotImplementedError
+                       raise OpenContourError(
+                               "Green theorem is not defined on open contours."
+                       )

        def _lineTo(self, p1):
                x0,y0 = self._getCurrentPoint()
```

**Pull request merged**